### PR TITLE
feat(EMS-659): Insurance page not found, get-application middleware

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds a new page not found  page for the insurance journey. When an application does not exist (e.g, invalid URL /reference number), we now redirect to this new 404 page instead of  "problem with service page.

## Summary
- Create new "page not found" route and controller
- Create new `get-application` middleware to fetch an application in all  /`insurance/:referenceNumber` routes.
  - If the application does not exist, redirect to the new 404 page
  - If the application exists, go to the requested page, passing the application via `res.locals`.
- Update/simplify all insurance controllers to consume `res.locals.application` instead of calling the API in every get/post controller.
- Simplify "save policy and export" function so that it consumes an application passed from the controller, instead of getting the application from the API.

## Other changes
- Update "your-business > company details" to consume and check if an application exists.
- Update "your-business > company details" route to have an application reference number.
- Update "your-business > company details" to generate page variables dynamically with a reference number, so that the post/redirect URLs are correct.
- Extract and rename "your-business > company details" routes and constants.

### CODE REVIEW

**Review by abhi-markan — APPROVED** [review]
LGTM

File: src/ui/server/controllers/insurance/all-sections/index.ts [hunk 1]
```diff
@@ -2,40 +2,31 @@ import { PAGES } from '../../../content-strings';
...
 import generateGroupsAndTasks from '../../../helpers/task-list/generate-groups-and-tasks';
 import generateTaskList from '../../../helpers/task-list';
 import flattenApplicationData from '../../../helpers/flatten-application-data';
 
 export const TEMPLATE = TEMPLATES.INSURANCE.ALL_SECTIONS;
 
-export const get = async (req: Request, res: Response) => {
-  const { referenceNumber } = req.params;
+export const get = (req: Request, res: Response) => {
+  const { application } = res.locals;
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/all-sections/index.ts:12]: should it be res?

File: src/ui/server/controllers/insurance/policy-and-export/save-data/index.ts [hunk 2]
```diff
@@ -1,56 +1,46 @@
...
-      // sanitise the form data.
-      const sanitisedData = sanitiseData(dataToSave);
+  if (application) {
+    // sanitise the form data.
+    const sanitisedData = sanitiseData(dataToSave);
 
-      // send the form data to the API for database update.
-      const policyAndExportId = application.policyAndExport.id;
+    // send the form data to the API for database update.
+    const policyAndExportId = application.policyAndExport.id;
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy-and-export/save-data/index.ts:32]: ` application.policyAndExport?.id;` <= incase policyAndExport does not exist?

File: src/ui/server/controllers/insurance/all-sections/index.ts (hunk 1 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/all-sections/index.ts:12]: yes 😄 the new middleware passes the application via `res.locals`

File: src/ui/templates/insurance/page-not-found.njk [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+{% block pageTitle %}
+  {{ CONTENT_STRINGS.PAGE_TITLE }}
+{% endblock %}
+
+{% block content %}
+
+  <h1 class="govuk-heading-l" data-cy="heading">{{ CONTENT_STRINGS.PAGE_TITLE }}</h1>
+
+  <p class="govuk-body" data-cy="body-1">{{ CONTENT_STRINGS.BODY_1 }}</p>
```
Comment (Zainzzkk) [inline src/ui/templates/insurance/page-not-found.njk:11]: I wonder if it's better making the name more descriptive then BODY_1 and BODY_2? not a blocker

Comment (ttbarnes) [reply src/ui/templates/insurance/page-not-found.njk:11]: Great shout, renamed to `TYPED_ADDRESS` and `PASTED_ADDRESS`

**Review by Zainzzkk — COMMENTED** [review]
failing tests

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Pull request merged**